build/flaws.js

```diff
@@ -55,10 +109,101 @@ function injectFlaws(doc, $) {
         }
       }
     });
-    if (options.flawLevels.get("bad_bcd_queries") === FLAW_LEVELS.ERROR) {
+    if (options.flawLevels.get("broken_links") === FLAW_LEVELS.ERROR) {
       throw new Error(`bad_bcd_queries flaws: ${doc.flaws.bad_bcd_queries}`);
     }
   }
 }
 
-module.exports = { injectFlaws };
+function fixFixableFlaws(doc, options, document) {
+  if (!options.fixFlaws || document.isArchive) return;
+
+  let newRawHTML = document.rawHtml;
+
+  const loud = options.fixFlawsDryRun || options.fixFlawsVerbose;
+
+  // Any 'macros' of type "MacroRedirectedLinkError"...
+  for (const flaw of doc.flaws.macros || []) {
+    if (
+      flaw.name === "MacroRedirectedLinkError" &&
+      (!flaw.filepath || flaw.filepath === document.fileInfo.path)
+    ) {
+      // Sanity check that our understanding of flaws, filepaths, and sources
+      // work as expected.
+      if (!newRawHTML.includes(flaw.macroSource)) {
+        throw new Error(
+          `rawHtml doesn't contain macroSource (${flaw.macroSource})`
+        );
+      }
+      const newMacroSource = flaw.macroSource.replace(
+        flaw.redirectInfo.current,
+        flaw.redirectInfo.suggested
+      );
+      // Remember, in JavaScript only the first occurrence will be replaced.
+      newRawHTML = newRawHTML.replace(flaw.macroSource, newMacroSource);
+      if (loud) {
+        console.log(
+          chalk.grey(
+            `Fixed macro ${chalk.white.bold(
+              flaw.macroSource
+            )} to ${chalk.white.bold(newMacroSource)}`
+          )
+        );
+      }
+      flaw.fixed = true;
```

Comment:
This little baby is important. Without it, you can't get an insight into what just happened. 
The fixable flaws do two things actually:
* Literally apply the suggestions in the flaws. 
* (optionally) rewrite the source

So, if you view a page, (with `BUILD_FIX_FLAWS=true` on) it will fix flaws so that what you're viewing it is already fixed (i.e. right-click and "Inspect element" on the links relevant) but it can also say what got fixed. 
If you have `BUILD_FIX_FLAWS_DRY_RUN=false`, simply refreshing the browser when on this page will entirely make all the (fixable) flaws go away.